File: queue/queue_array_implementation.py
# not recomended, use LL for queue implementations instead, it's more performant
class Queue:
    def __init__(self):
        self.list = []

    def enqueue(self, val):
        self.list.append(val)
        return self.list

    def dequeue(self):
        self.list.pop(0)
        return self.list


queue = Queue()
print(queue.enqueue(1))  # [1]
print(queue.enqueue(2))  # [1, 2]
print(queue.dequeue())  # [2]

File: queue/_old/queue-array-implementation.js
// Not recomended, use LL for queue implementations instead
class Queue {
  constructor() {
    this.arr = [];
  }

  add(val) {
    return this.arr.push(val);
  }

  remove() {
    return this.arr.shift();
  }
}

const q = new Queue();
q.add(1); // 1
q.add(2); // 2
q.remove(); // 1
